### ISSUE TITLE
e2e/build/docker flake: Successfully built X, no such id: sha256:<X>

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kubernetes#21991]
I'm guessing a concurrent docker cleanup and I just got really lucky/unlucky. Although I think running this in a container might be new?

https://github.com/kubernetes/kubernetes/pull/21905#issuecomment-188413640

```
Step 7 : ENV KUBE_OUTPUT_SUBPATH _output/dockerized
 ---> Running in 381efd9d8294
 ---> 5c0ca2e27aba
Removing intermediate container 381efd9d8294
Step 8 : ADD kube-source.tar.gz /go/src/k8s.io/kubernetes/
 ---> d61e1b774b59
Removing intermediate container ef7273a596a9
Successfully built d61e1b774b59
no such id: sha256:d61e1b774b595c6d0b2c3d94c07554195a8c5244ab16e7e007243530f6145e83

To retry manually, run:

docker build -t kube-build:build-2be8cc7bdc --pull=false /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/_output/images/kube-build:build-2be8cc7bdc

!!! Error in /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:531
  'return 1' exited with status 1
Call stack:
  1: /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:531 kube::build::build_image(...)
  2: /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/release.sh:31 main(...)
Exiting with status 1
2016/02/24 11:17:18 e2e.go:278: Error running build-release: exit status 1
```

**Comment by ixdy** [comment]
Docker changed the format of ids in 1.10 to include the `sha256:`, so it's possible some of our infra doesn't work with the new format. (See https://issues.jenkins-ci.org/browse/JENKINS-32792 and https://github.com/docker/docker/pull/17924.)

We've since reverted to Docker 1.9.1 on Jenkins, so this shouldn't recur for now. We'll probably need to figure out if something in the build infra needs to be fixed at some point, x-ref #19720, but I don't have time to investigate right now.

cc @kubernetes/sig-node @david-mcmahon

**Comment by fejta** [comment]
Let's punt this until we're actually using 1.10 again
